### PR TITLE
use storage multi_get

### PR DESCRIPTION
**Pull request by nkysg**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
use multi_get replace for loop

-
-
-

### CODE REVIEW

**Comment by nkysg** [comment]
list_block 需要修改下

File: chain/service/src/chain_service.rs [hunk 1]
```diff
@@ -298,6 +301,7 @@ impl ReadableChainService for ChainReaderServiceInner {
         self.storage.get_blocks(ids)
     }
 
+    // XXX FIXME different get_blocks
```
Comment (nkysg) [inline chain/service/src/chain_service.rs:304]: 这里风格使用Vec<T>，其他地方用的是Vec<Option<T>>不统一

File: chain/src/chain.rs [hunk 2]
```diff
@@ -746,19 +746,20 @@ impl ChainReader for BlockChain {
...
-            let info = self.storage.get_transaction_info(hash)?.ok_or_else(|| {
-                anyhow::anyhow!(format!(
-                    "cannot find hash({}) on main chain(head: {})",
-                    hash,
+        let txn_infos = self.storage.get_transaction_infos(hashes.clone())?;
+        for (i, info) in txn_infos.into_iter().enumerate() {
+            match info {
+                Some(info) => infos.push(info),
+                None => bail!(
+                    "cannot find hash({:?}) on main chain(head: {})",
```
Comment (jolestar) [inline chain/src/chain.rs:758]: 这里 main chain 可能有点歧义

**Review by jolestar — APPROVED** [review]
LGTM

File: chain/src/chain.rs (hunk 2 above)
Comment (nkysg) [reply chain/src/chain.rs]: fix it

File: chain/src/chain.rs [hunk 3]
```diff
@@ -566,14 +566,16 @@ impl ChainReader for BlockChain {
...
-                    .ok_or_else(|| format_err!("Can not find block by number {}", idx))
-            })
-            .collect()
+        let ids = self.get_block_ids(end_num, true, count)?;
+        let block_opts = self.storage.get_blocks(ids)?;
+        let mut blocks = vec![];
+        for (idx, block) in block_opts.into_iter().enumerate() {
+            match block {
+                Some(block) => blocks.push(block),
+                None => bail!("Can not find block by number {}", idx),
```
Comment (jolestar) [inline chain/src/chain.rs:575]: 这里的 idx 是 enumerate 的 index 吧？不是 block number 吧？

Comment (nkysg) [reply chain/src/chain.rs:575]: 对，我修改下

**Review by jolestar — APPROVED** [review]

**Pull request merged**